test/python/tools/test_compiler.py

```diff
@@ -342,6 +341,47 @@ def test_compile_pass_manager(self):
         rfalse = backend.run(qrfalse).result()
         self.assertEqual(rtrue.get_counts(), rfalse.get_counts())
 
+    def test_compile_with_initial_layout(self):
+        """Test compile with an initial layout.
+        Regression test for #1711
+        """
+        qr = QuantumRegister(3)
+        cr = ClassicalRegister(3)
+        qc = QuantumCircuit(qr, cr)
+        qc.u1(3.14, qr[0])
+        qc.cx(qr[1], qr[0])
+        qc.barrier(qr)
+        qc.measure(qr, cr)
+        backend = FakeRueschlikon()
+        initial_layout = {0: (qr, 1), 2: (qr, 0), 15: (qr, 2)}
```

Comment:
I propose that this test should be modified. As it is, the circuit has a `cx qr[1], qr[0]`, and the initial_layout says that should become `cx 0, 2`. The FakeRueschlikon coupling_map is:

```
        1  -  2  -  3  -  4  -  5  -  6  -  7   -  8

        |     |     |     |     |     |     |      |

        0  -  15 -  14 -  13 -  12 -  11 -  10  -  9
```
so then I don't understand why this gets translated to a `cx 15, 0` and `cx 15, 2`.

It seems this relies on the underlying Swap algorithm, which may be different depending on version of qiskit.

I think this test should be much simpler, and not require any swap. I would just do the following:
```
qc.cx(qr[2], qr[1])
qc.cx(qr[2], qr[0])

initial_layout = {0: (qr, 1), 2: (qr, 0), 15: (qr, 2)}
```
Then this circuit should run with no modification (no extra gate). So the assert can be that the circuit still has the same type of gates.

Reply:
excellent point. fixed in e6a5c62a